models/oauth2_application.go

```diff
@@ -431,6 +435,36 @@ func (grant *OAuth2Grant) increaseCount(e Engine) error {
 	return nil
 }
 
+// ScopeContains returns true if the grant scope contains the specified scope
+func (grant *OAuth2Grant) ScopeContains(scope string) bool {
+	for _, currentScope := range strings.Split(grant.Scope, " ") {
+		if scope == currentScope {
+			return true
+		}
+	}
+	return false
+}
+
+// SetNonce updates the current nonce value of a grant
+func (grant *OAuth2Grant) SetNonce(nonce string) error {
+	return grant.setNonce(x, nonce)
+}
+
+func (grant *OAuth2Grant) setNonce(e Engine, nonce string) error {
+	newGrant := new(OAuth2Grant)
+	newGrant.Nonce = nonce
+	_, err := e.ID(grant.ID).Cols("nonce").Update(newGrant)
+	if err != nil {
+		return err
+	}
+	updatedGrant, err := getOAuth2GrantByID(e, grant.ID)
+	if err != nil {
+		return err
+	}
+	grant.Nonce = updatedGrant.Nonce
```

Comment:
not sure I completely understand this - update should lock the table so the result updatedGrant.Nonce must be nonce

Reply:
@zeripath I agree, since I am not very familiar with this codebase (yet) I simply attempted to mimic what was done elsewhere, but I see now that it was unnecessary.  There is no need to do a readback here, so I have updated it.